superset/translations/utils.py

```diff
@@ -38,7 +38,7 @@ def get_language_pack(locale):
         try:
             with open(filename, encoding="utf8") as f:
                 pack = json.load(f)
-                ALL_LANGUAGE_PACKS[locale] = pack
+                ALL_LANGUAGE_PACKS[locale] = pack or {}
         except Exception:  # pylint: disable=broad-except
             # Assuming english, client side falls back on english
```

Comment:
Perhaps we should put:
```python
pack = {}
```
in this block, and move
```python
ALL_LANGUAGE_PACKS[locale] = pack
```
outside of the try/except.

Then we could remove the `Optional` from the `get_language_pack` return type

Reply:
@etr2460 I thought of that however it would change the return type, i.e., it could no longer be `None` and per the comment it seems like this could be problematic for the client.

Reply:
Doesn't an empty dict resolve to false? so if the locale is English then this returns an empty dict (based on the constant at the top of the file)

Reply:
And this seems to indicate that if this isn't set, we default to an empty object on the frontend anyway: https://github.com/apache-superset/superset-ui/blob/master/packages/superset-ui-translation/src/Translator.ts#L26

Reply:
@etr2460 the idea here was to keep the status quo. `pack` could be `None` per line 35 and remain that way if an exception is thrown either when opening the file or parsing the JSON.

Reply:
I'm still not 100% convinced that it's not worth cleaning up a bit and making this not `Optional`, but if you don't want to make such a change in the typing PR, that's fine. I can make a follow up PR to update this logic as I think it's cleaner